### DISCUSSION TITLE
Questions about Incremental Static Regeneration

### DISCUSSION THREAD

**Discussion by MehediH** [vercel/next.js#17001] [Help] [answered]
Hi everyone,

I am playing around with Incremental Static Regeneration on one of my projects where one of the main goals is to minimise calls to the database/API. 

I have an Edit functionality on the project that allows users to update an entry on the database and then view the updated version on the Next.js app (which uses Incremental Static Regeneration). So ideally, I would only want to re-generate that page when a user actually updates the entry corresponding to that page on the database. 

Is there any way to "manually" re-generate specific pages, instead of re-building the entire site or incrementally regenerating with something like `revalidate`? 

I am not entirely sure what the best path here would be, and I would appreciate any suggestions! 

Thank you :)

**Comment by leerob** [comment, marked as answer]
There currently isn't a way to programmatically revalidate pages via something like a webhook or API. In the meantime, one option would be to serve a static loading state from the server followed by client-side rendering. 

1. Your Next.js app renders loading state from the server, so the first paint is fast
2. Then, you make an API call using a library like [SWR](https://swr.vercel.app/)
3. SWR allows you to reduce the number of calls to your database by only fetching when the content is stale
4. A user edits an entry in the database
5. You can manually update the SWR cache instantly, and then refetch in the background

Example of an edit modal which mutates an SWR cache: https://github.com/leerob/fastfeedback/blob/master/components/EditSiteModal.js#L38

**Reply by leerob** [reply]
An on-demand invalidation API is in the works, but not ready yet.

https://github.com/vercel/next.js/discussions/11552#discussioncomment-52146
